mkosi/config.py

```diff
@@ -905,23 +905,22 @@ def config_default_repository_key_fetch(namespace: argparse.Namespace) -> bool:
     if detect_distribution()[0] != Distribution.ubuntu:
         return False
 
+    def needs_repository_key_fetch(distribution: Distribution) -> bool:
+        return distribution == Distribution.arch or distribution.is_rpm_distribution()
+
     if namespace.tools_tree is None:
-        return cast(bool, namespace.distribution.is_rpm_distribution())
+        return needs_repository_key_fetch(namespace.distribution)
 
     if namespace.tools_tree != Path("default"):
         return (
             detect_distribution(namespace.tools_tree)[0] == Distribution.ubuntu
-            and namespace.distribution.is_rpm_distribution()
-        )
+            and needs_repository_key_fetch(namespace.distribution)
+        )  # fmt: skip
```

Comment:
Wow the way it reformats this is insane, that can only be a bug in ruff. So it can be referenced in a bug report (ruff version 0.9.1):
```diff
         return needs_repository_key_fetch(namespace.distribution)
 
     if namespace.tools_tree != Path("default"):
-        return (
-            detect_distribution(namespace.tools_tree)[0] == Distribution.ubuntu
-            and needs_repository_key_fetch(namespace.distribution)
-        ) 
+        return detect_distribution(namespace.tools_tree)[
+            0
+        ] == Distribution.ubuntu and needs_repository_key_fetch(namespace.distribution)
 
     return (
         namespace.tools_tree_distribution == Distribution.ubuntu
```